x-pack/legacy/plugins/maps/index.js

```diff
@@ -118,6 +116,13 @@ export function maps(kibana) {
           elasticsearch: server.plugins.elasticsearch,
         },
         savedObjects: {
+          savedObjectsClient: (() => {
+            const callCluster = server.plugins.elasticsearch.getCluster('admin')
+              .callWithInternalUser;
+            const { SavedObjectsClient, getSavedObjectsRepository } = server.savedObjects;
+            const internalRepository = getSavedObjectsRepository(callCluster);
+            return new SavedObjectsClient(internalRepository);
+          })(),
```

Comment:
no need for self-executing function imho. it adds nesting.

Just move this block up to the `init`-scope and assign with `savedObjectsClient: new SavedObjectClients(internalRepository)`